src/eei.cpp

```diff
@@ -57,6 +57,83 @@ Literal EthereumInterface::callImport(Import *import, LiteralList& arguments) {
       return Literal();
     }
 
+    if (import->base == Name("getBalance")) {
+      std::cout << "getbalance\n";
+
+      uint32_t addressOffset = arguments[0].geti32();
+      uint32_t resultOffset = arguments[1].geti32();
+
+      evm_address *address = new evm_address;
+      evm_uint256be *result = new evm_uint256be;
+
+      loadUint160(addressOffset, address);
+      hera.context->fn_table->get_balance(result, hera.context, address);
+      storeUint128(resultOffset, result);
+
+      delete address;
+      delete result;
+
+      return Literal();
+    }
+
+    if (import->base == Name("getBlockHash")) {
+      std::cout << "getblockhash ";
+
+      int64_t number = arguments[0].geti64();
+      uint32_t resultOffset = arguments[1].geti32();
+
+      std::cout << number << "\n";
+
+      evm_uint256be *blockhash = new evm_uint256be;
+      hera.context->fn_table->get_block_hash(blockhash, hera.context, number);
+      storeUint256(resultOffset, blockhash);
+
+      delete blockhash;
+
+      return Literal();
+    }
+
+    if (import->base == Name("call")) {
+      std::cout << "call\n";
+
+      int64_t gas = arguments[0].geti64();
+      uint32_t addressOffset = arguments[1].geti32();
+      uint32_t valueOffset = arguments[2].geti32();
+      uint32_t dataOffset = arguments[3].geti32();
+      int32_t dataLength = arguments[4].geti32();
+      uint32_t resultOffset = arguments[5].geti32();
+      int32_t resultLength = arguments[6].geti32();
+
+      evm_result *call_result = new evm_result;
+      evm_message *call_message = new evm_message;
```

Comment:
I think we should have a helper for building a `call_message`, it is going to be shared by all the different call implementations.

Reply:
this is a good idea, since callCode and other interface methods pretty much repeat all of this